### ISSUE TITLE
Missing icons on bar when temporary folder doesn't exists

### ISSUE THREAD

**Issue by jmiguelr** [artikell/joplin-excalidraw#10]
I installed your plugin yesterday and at first sigth it didn't work at all: no icon at any place. 

After some research, I saw a message stating that looking at the developer tools page there could be an error about a missing tmp folder. And bingo, that it was. After creating (in my case) /tmp/joplin-excalidraw-plugin everything looks working well (apart from the double-click issue reported at another ticket, 

Environment: Linux Mint 21.2 with Cinnamon UI . Joplin 2.12.15,  Plugin version: 1.3.0

Thank you very much for your work!

**Comment by salehawal** [comment]
i have created the temp folder but still not working
sudo mkdir /tmp/joplin-excalidraw-plugin

any help?

**Comment by robertohuertasm** [comment]
@salehawal, this is what I did:
- I'm using a Mac
- I executed node in a terminal
- then `os.tmpdir()`
- the output was something like `/var/folders/wk/7gc3q6pd6p573wttw34t8lsr0000gq/T`
- create a folder called `joplin-excalidraw-plugin` in the previous directory
- restart joplin
- you should see the icon

**Comment by salehawal** [comment]
still not working, even after creating the temp folder "/tmp/joplin-excalidraw-plugin" @jmiguelr 

using ubuntu linux as host

**Comment by salehawal** [comment]
> @salehawal, this is what I did:
> 
>     * I'm using a Mac
> 
>     * I executed node in a terminal
> 
>     * then `os.tmpdir()`
> 
>     * the output was something like `/var/folders/wk/7gc3q6pd6p573wttw34t8lsr0000gq/T`
> 
>     * create a folder called `joplin-excalidraw-plugin` in the previous directory
> 
>     * restart joplin
> 
>     * you should see the icon

yes the icon is visible now, but after creating the image and saving it, it will now be displayed on the draw, only exceldraw icons shows up
![Screenshot from 2024-03-22 00-54-35](https://github.com/artikell/joplin-excalidraw/assets/3238761/c829b650-edc5-401f-af82-e0b870061443)


and after trying to edit the saved image, the imported image in the drawing is not there anymore?
![Screenshot from 2024-03-22 00-54-52](https://github.com/artikell/joplin-excalidraw/assets/3238761/60eacdbb-f198-43ef-9c61-5595e190e6cc)

**Comment by djsudduth** [comment]
Same issue here - missing icon- Windows 10, Joplin 2.14.19, Excalidraw 1.4.1.  Had to create joplin-excalidraw-plugin folder in my Windows temp directory. Go to the command line - type `set` to find your temp path. 

Mine was
`C:\Users\(myusername)\AppData\Local\Temp` 

Create a folder -> `C:\Users\(myusername)\AppData\Local\Temp\joplin-excalidraw-plugin`

This will fix it

**Comment by Paulswith** [comment]
arch linux:

`mkdir /tmp/joplin-excalidraw-plugin `
restart Joplin,
works

**Comment by gr4phite** [comment]
I have the same issue as @salehawal, on NixOS with the Joplin flatpak. I created the temporary directory, but only the Excalidraw logo gets inserted. Though perhaps this is unrelated..

**Comment by tekjunkie6502** [comment]
I seem to have the same issue on linux mint 21.3, now running joplin 3.0.13
Joplin 3.0.13 (prod, linux)
joplin-excalidraw: 1.3.0

the "mkdir /tmp/joplin-excalidraw-plugin " didn't work for me

**Comment by Fricsion** [comment]
> @salehawal, this is what I did:
> 
>     * I'm using a Mac
> 
>     * I executed node in a terminal
> 
>     * then `os.tmpdir()`
> 
>     * the output was something like `/var/folders/wk/7gc3q6pd6p573wttw34t8lsr0000gq/T`
> 
>     * create a folder called `joplin-excalidraw-plugin` in the previous directory
> 
>     * restart joplin
> 
>     * you should see the icon

This worked thank you

**Comment by bfcns** [comment]
> I seem to have the same issue on linux mint 21.3, now running joplin 3.0.13 Joplin 3.0.13 (prod, linux) joplin-excalidraw: 1.3.0
> 
> the "mkdir /tmp/joplin-excalidraw-plugin " didn't work for me

uninstall and reinstall the plugin

**Comment by qlixed** [comment]
Here an example of what I see on my Joplin flatpak install:

<img width="725" height="406" alt="Image" src="https://github.com/user-attachments/assets/ae437397-2be6-4b1c-8cf7-96be3005ae9d" />

If you use flatpak you need to make the dir inside /run that is the environment where tmp is mapped for joplin running as flatpak:

mkdir /run/user/$UID/app/net.cozic.joplin_desktop/joplin-excalidraw-plugin

$UID is the UID of your user, usually is an env var set on any linux distribution. After this you need to restart joplin and it should work.
But this is something that need to be fixed from the app, if the directory don't exist it should be created, as /run on some systems (e.g: fedora) is a tmpfs and after reboot the created directory will be deleted. In the code it looks like is created if it doesn't exists:

https://github.com/artikell/joplin-excalidraw/blob/v1.4.1/src/resources.ts#L24

But that don't happen properly apparently

**Comment by santakdalai90** [comment]
> Here an example of what I see on my Joplin flatpak install:
> <img alt="Image" width="725" height="406" src="https://private-user-images.githubusercontent.com/1832112/467050807-ae437397-2be6-4b1c-8cf7-96be3005ae9d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkzODM4NzcsIm5iZiI6MTc1OTM4MzU3NywicGF0aCI6Ii8xODMyMTEyLzQ2NzA1MDgwNy1hZTQzNzM5Ny0yYmU2LTRiMWMtOGNmNy05NmJlMzAwNWFlOWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDJUMDUzOTM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWQ2NDM0ODkxMzUyZTc0ZmVhY2M0YTE2Zjk2NmY2NDQ3OGFlNzY4M2JjYWIyNDFjNmE1M2ViMDIzMmVlOGEyNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FVz4qy4RNwwyrlxss3oAB9MdqKKnPl83onfteMGlazM">
> 
> If you use flatpak you need to make the dir inside /run that is the environment where tmp is mapped for joplin running as flatpak:
> 
> mkdir /run/user/$UID/app/net.cozic.joplin_desktop/joplin-excalidraw-plugin
> 
> $UID is the UID of your user, usually is an env var set on any linux distribution. After this you need to restart joplin and it should work. But this is something that need to be fixed from the app, if the directory don't exist it should be created, as /run on some systems (e.g: fedora) is a tmpfs and after reboot the created directory will be deleted. In the code it looks like is created if it doesn't exists:
> 
> https://github.com/artikell/joplin-excalidraw/blob/v1.4.1/src/resources.ts#L24
> 
> But that don't happen properly apparently

If you've installed it using snap on a ubuntu based system then it should be 

```bash
mkdir /run/user/$UID/snap.joplin-desktop/joplin-excalidraw-plugin 
```
